modules/nodes-cluster-resource-override-move-infra.adoc

```diff
@@ -0,0 +1,188 @@
++
+[source,terminal]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: clusterresourceoverride
+  namespace: clusterresourceoverride-operator
+# ...
+spec:
+  config:
+    nodeSelector:
+      node-role.kubernetes.io/infra: "" <1>
+# ...
+----
+<1> Specify the role of the node where you want to deploy the Cluster Resource Override Operator pod.
++
+[NOTE]
+====
+If the infra node uses taints, you need to add a toleration to the `Subscription` CR.
+
+For example:
+
+[source,terminal]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: clusterresourceoverride
+  namespace: clusterresourceoverride-operator
+# ...
+spec:
+  config:
+    nodeSelector:
+      node-role.kubernetes.io/infra: ""
+    tolerations: <1>
+    - key: "node-role.kubernetes.io/infra"
+      operator: "Exists"
+      effect: "NoSchedule"
+----
+<1> Specifies a toleration for a taint on the infra node.
+====
+
+. Move the Cluster Resource Override pods by adding a node selector to the `ClusterResourceOverride` custom resource (CR):
+
+.. Edit the CR:
++
+[source,terminal]
+----
+$ oc edit ClusterResourceOverride cluster -n clusterresourceoverride-operator
+----
+
+.. Add a node selector to match the node role label on the infra node:
++
+[source,terminal]
+----
+apiVersion: operator.autoscaling.openshift.io/v1
+kind: ClusterResourceOverride
+metadata:
+  name: cluster
+  resourceVersion: "37952"
+spec:
+  podResourceOverride:
+    spec:
+      cpuRequestToLimitPercent: 25
+      limitCPUToMemoryPercent: 200
+      memoryRequestToLimitPercent: 50
+  deploymentOverrides:
+    replicas: 1 <1>
+    nodeSelector:
+      node-role.kubernetes.io/infra: "" <2>
+# ...
+----
+<1> Optional: Specify the number of Cluster Resource Override pods to deploy. The default is 2. Only one pod is allowed per node.
+<2> Optional: Specify the role of the node where you want to deploy the Cluster Resource Override pods.
++
+[NOTE]
+====
+If the infra node uses taints, you need to add a toleration to the `ClusterResourceOverride` CR.
+
+For example:
+
+[source,terminal]
+----
+apiVersion: operator.autoscaling.openshift.io/v1
+kind: ClusterResourceOverride
+metadata:
+  name: cluster
+# ...
+spec:
+  podResourceOverride:
+    spec:
+      memoryRequestToLimitPercent: 50
+      cpuRequestToLimitPercent: 25
+      limitCPUToMemoryPercent: 200
+  deploymentOverrides:
+    replicas: 3
+    nodeSelector:
+      node-role.kubernetes.io/worker: ""
+    tolerations: <1>
+    - key: "key"
+      operator: "Equal"
+      value: "value"
+      effect: "NoSchedule"
+----
+<1> Specifies a toleration for a taint on the infra node.
+====
+
+.Verification
+
+* You can verify that the pods have moved by using the following command:
++
+[source,terminal]
+----
+$ oc get pods -n clusterresourceoverride-operator -o wide
+----
++
+The Cluster Resource Override pods are no longer deployed to the control plane nodes.
```

Comment:
I think this should say `no longer deployed to the worker nodes` if we are still following the example from above where the pods were scheduled to the `worker` nodes. In your example, they are now scheduled to the `infra,worker` nodes (which are technically still worker nodes).

If you don't want to recreate this all over again, we can probably just manually adjust the names in this example so that it looks like they were actually deployed to control plane nodes: e.g.
> The following examples shows the Cluster Resource Override pods and the Cluster Resource Override Operator pod deployed to ~worker~ control plane nodes.

and in the `Example Cluster Resource Override pods` example right below that line, change to:
```
NAME                                                READY   STATUS    RESTARTS   AGE   IP            NODE                                        NOMINATED NODE   READINESS GATES
clusterresourceoverride-786b8c898c-9wrdq            1/1     Running   0          23s   10.131.0.33   ip-10-0-2-39.us-west-2.compute.internal   <none>           <none>
clusterresourceoverride-786b8c898c-vn2lf            1/1     Running   0          26s   10.131.0.33   ip-10-0-2-39.us-west-2.compute.internal   <none>           <none>
clusterresourceoverride-operator-6b8b8b656b-lvr62   1/1     Running   0          56m   10.131.0.33   ip-10-0-2-39.us-west-2.compute.internal   <none>           <none>
```

Reply:
@mburke5678, do you want to address this new comment before getting this PR peer-reviewed?

Reply:
Good point, @max-cx . Looks like a small change is needed in the example.


![Screenshot from 2024-09-20 09-54-53](https://github.com/user-attachments/assets/84b51d6a-57c8-4f6f-9a9a-2a36cc58fab6)

Reply:
@maxcao13 In Slack you told me: 

> the operator pod itself clusterresourceoverride-operator-### pod will automatically deploy to a worker node, ...and clusterresoureceoverride-### pods will automatically be scheduled to noderole=master nodes

So I now have 

```
NAME                                                READY   STATUS    RESTARTS   AGE   IP            NODE                                        NOMINATED NODE   READINESS GATES
clusterresourceoverride-786b8c898c-9wrdq            1/1     Running   0          23s   10.128.2.32   ip-10-0-14-183.us-west-2.compute.internal   <none>           <none>
clusterresourceoverride-786b8c898c-vn2lf            1/1     Running   0          26s   10.130.2.10   ip-10-0-20-140.us-west-2.compute.internal   <none>           <none>
clusterresourceoverride-operator-6b8b8b656b-lvr62   1/1     Running   0          56m   10.131.0.33   ip-10-0-2-39.us-west-2.compute.internal     <none>           <none>
```
where `10-0-14-183` and `10-0-20-140` are control plane and `10-0-2-39` is a worker.

Reply:
Yes, the default should be what I said, so what you wrote above is okay to keep since that will be the state at the beginning of this section. And then the resulting state after applying `deploymentOverrides`, should be  the 2 `clusterresourceoverride-786b8c898c-###` pods on worker nodes or infra nodes.

But right now in your example, right below my this comment, the 2 pods are still placed the exact same as the original state.

Reply:
@maxcao13 Ah, I didn't realize that the pod IP also changes when the pod is moved. Seems so obvious now! Thank you for all your help.